### ISSUE TITLE
Limit for string.split method

### ISSUE THREAD

**Issue by QuentinCaffeino** [godotengine/godot#14349]
What if you don't want to split full string? Maybe you need only first 3 parts?

If limit is set and positive, the returned array will contain a maximum of limit elements with the last element containing the rest of string.

If the limit parameter is negative, all components except the last -limit are returned.

eg.:
```gdscript
var some_string = "Lorem ipsum dolor sit amet, consectetur adipiscing elit."

some_string.split(" ", false, 3)  # [ "Lorem", "ipsum", "dolor sit amet, consectetur adipiscing elit."]
some_string.split(" ", false, -2)  # [ "Lorem", "ipsum", "dolor", "sit", "amet,", "consectetur"]
```

**Comment by willnationsdev** [comment]
> if limit is set and positive

It should probably have a default parameter value of `-1` so that it's always set and negative by default. That way we can avoid compatibility issues with existing function calls. Idk if that's what you meant, but...

**Comment by QuentinCaffeino** [comment]
@willnationsdev if we would use `-1` by default it will return all words except the last one.
I think max int `std::numeric_limits<int>::max()` (maybe godot has its own implementation for this val some where in headers) is ok for default value)

PS. We don't have to think of our way of doing this. It is okay to look at other implementations in other languages :)

**Comment by bojidar-bg** [comment]
Well, we can go with a default of `(-)0`, so it would return all elements except the last 0.

Maybe for negative numbers, we should concatenate the last N elements, instead of dropping them?

**Comment by QuentinCaffeino** [comment]
@bojidar-bg well it is good going with max value by default so there wouldn't be unexpected overflow. If someone knows what he is doing he could use `0` to ''disable limits''.

I think it will add unneeded logic to code. Eg.: you would want to print all words you got from split, instead of writing:
```gdscript
for word in words:
  print(word)
```
You will be doing:
```gdscript
for i in range(limit):  # Cause you don't need last concatenated part
  print(words[i])
```
And the logic gets even worse if you trying to print words from `rsplit` method.

Also I think `rsplit` method should be added. The same but from the right.

**Comment by Calinou** [comment]
This was implemented by https://github.com/godotengine/godot/pull/14580 a while ago (which makes it available in Godot 3.0 onwards), closing.
